pages/base.py

```diff
@@ -50,30 +50,32 @@ class BasePage(BaseElement):
     def open(self):
         self.driver.get(self.url)
 
+    def verify_page(self):
+        """
+        Use an element from the locator_dictionary to wait for the page to load,
+        then checks if browser is at expected URL.
+        """
+        # self.__getattr__(list(self.locator_dictionary.keys())[0])
+        return self.url.rstrip('/') == self.driver.current_url.rstrip('/')
```

Comment:
verify_page should default to return True, then in each page overwrite this function to its own version. Instead of check the url, check the unique html elements on that page to verify the page is correct page.

Reply:
Would it make any sense to have a default one that just pulls an attribute (or a couple) from the locator dictionary and check if they're there? This would not always work because locators have the possibility to change and could mean different things on different pages, but if you check a couple it would be unlikely you're on the wrong page and if the locators are broken we would want to know that anyway.

Reply:
If anything on that page changes and break the tests, then someone need to fix that. I think verify page with the elements instead of url is more ideal for browser testing. You can always check urls in all the unit tests in osf.

Reply:
Makes sense. Does that default I suggested make sense, then?

Reply:
That sounds good.